### ISSUE TITLE
Shutting down fleet can leave a step inconsistent

### ISSUE THREAD

**Issue by donkirkby** [cfe-lab/Kive#532]
The sample `N61656A-MIDI` was run once, and the fleet crashed just as the outputs were being checked from the `aln2counts` step. It was run again, and something went wrong when it tried to fill in the exec record for the `aln2counts` step. It is currently running again.

**Comment by rhliang** [comment]
We did some digging, and believe we have found the problem that happened with that step:

* When the fleet crashed, the original run (R1) had its `aln2counts` step in progress.  However, the step itself had actually finished, but the system was busily performing a content check on the outputs.  It had already completed content checks on two of its outputs but the third content check was not finished (it had not had its `start_time` set yet even).  That left three outputs without content checks at all.
* On the second run (R2) that was restarted from the first, when it came to the `aln2counts` step, it found the execrecord created during R1.  This execrecord was usable but not fully reusable, so Kive re-ran the code as expected.
* When the code was finished running, it saw that three of the outputs had already received content checks, so went on to only perform content checks on the other three outputs *even though one of the existing ones was incomplete*.  At this point, the execrecord's 6 outputs all have content checks, 5 of which are fine, but 1 of which is incomplete.
* Then R2 attempted to run the output cable associated with the output that had the incomplete content check, at which point it complained about the dataset, and marked the run and the output cable as cancelled.

So, we need to change our execution code to not only look for an existent content check, but a content check that is *complete*.  This is line 1790 of `sandbox.execute`, as of commit c65881e8d25b5157614b9956ebc137f080b9a4f7.
